overlord/devicestate/devicemgr.go

```diff
@@ -954,6 +957,50 @@ func (m *DeviceManager) Ensure() error {
 	return nil
 }
 
+// withKeypairMgr invokes a function making the device KeypairManager
+// available to it.
+// It uses the right location for the manager depending on UC16/18 vs 20,
+// the latter uses ubuntu-save.
+// For UC20 it also checks that ubuntu-save is mounted.
+func (m *DeviceManager) withKeypairMgr(f func(asserts.KeypairManager) error) error {
+	// we use the model to check whether this is a UC20 device
+	// TODO: during a theoretical UC18->20 remodel the location of
+	// keypair manager keys would move, we will need dedicated code
+	// to deal with that, this code typically will return the old location
+	// until a restart
+	model, err := m.Model()
+	if err == state.ErrNoState {
+		return fmt.Errorf("internal error: cannot access device keypair manager before a model is set")
+	}
+	if err != nil {
+		return err
+	}
+	underSave := false
+	if model.Grade() != asserts.ModelGradeUnset {
+		// on UC20 the keys are kept under the save dir
+		underSave = true
+	}
+	where := dirs.SnapDeviceDir
+	if underSave {
+		// check for availability in case save gets mounted/unmounted,
+		// it's responsibility of the caller to have it mounted
+		if !m.saveAvailable {
```

Comment:
I don't understand your worry. I'm probably missing something. Given that /save can be there but unmounted we still need some state or otherwise the helper gets expensive. There's is not a lot of places that need to use/touch save hopefully. The idea is that they ill have a matching with* helper.